File: CPP/cow.cpp
#include <iostream>
using namespace std ;

#include "cow.h"
#include "cowmilk.h"
#include "cowmeat.h"

void Cow::render() {
	if (this->getHungry() ){
		cout << "s" ;
	}else{
		cout << "S" ;
	}
}

void Cow::ToSound() {
	cout << "Suara Cow" << endl ;
}

int Cow::getDurationForHungryTime() {
	return 25 ; 
}

FarmProduct* Cow::createProduct() {
	FarmProduct* p = new CowMilk() ;
	return p ;
}

FarmProduct* Cow::getMeatProduct() {
	FarmProduct* p = new CowMeat() ;
	return p ;
}

File: JAVA_noPackages/Cow.java
public class Cow extends FarmAnimal implements MilkProducingFarmAnimal, MeatProducingFarmAnimal{
    public boolean IsMeatProducing(){
        return true;
    }

    public boolean IsMilkProducing(){
        return true;
    }

    public void render() {
        if (this.getHungry() ){
            System.out.print("s");
        }else{
            System.out.print("S");
        }
    }
    
    public void ToSound() {
        System.out.println("Mooo");
    }
    
    public int getDurationForHungryTime() {
        return 25 ; 
    }
    
    public FarmProduct createProduct() {
        FarmProduct p = new CowMilk() ;
        return p ;
    }
    
    public FarmProduct getMeatProduct() {
        FarmProduct p = new CowMeat() ;
        return p ;
    }
    
    public Cow(){
		super.setMeatProducing(true);
	}
}
